python-lib/amazon_transcribe_api_client.py

```diff
@@ -318,3 +324,24 @@ def _result_parser(self,
             logging.warning(f"Unknown state encountered: {job_status}")
             raise UnknownStatusError(f"Unknown state encountered: {job_status}")
         return job_data
+
+    def check_job_timeout(self,
+                          job_summary: Dict,
+                          job_res_data: Dict):
+        """
+        If the job duration is longer than the timeout setup previously, a JOB_TIMEOUT_ERROR will
+        be written in the column error of the Dataframe, otherwise it returns None.
+        """
+
+        date_job_created = job_summary.get("CreationTime")
+        now = datetime.datetime.now(tz=date_job_created.tzinfo)
+        time_delta_min = int((now - date_job_created).seconds / 60)
+        logging.info(
+            f"{job_summary.get('TranscriptionJobStatus')} | {job_summary.get('TranscriptionJobName')} | {time_delta_min} min")
+        if self.use_timeout and time_delta_min >= self.timeout:
+            logging.error(f'Job {job_summary.get("TranscriptionJobName")} timeout!')
```

Comment:
This should rather be a warning than an error. I'd also change the text to be more informative: "Job {jobname} canceled after a timeout of {self.timeout} minutes"

We also need to cancel the amazon job when the timout happens

Reply:
done in 1b4d2379271f3e34715ff74b42eb6f17053e741a

ps: as we said before, we cannot cancel an "in progress" job